File: Frontend/src/interfaces/TransportCompanies.ts
import { TransportVehiclesInterface } from "./TransportVehicles";

export interface TransportCompaniesInterface {
  id?: number;
  company_name?: string;
  contact?: string;
  email?: string;
  address?: string;
  postal_code?: string;
  start_date?: string;  // Can use Date type if working with actual Date objects
  end_date?: string;  // Can use Date type if working with actual Date objects
  photo?: string;

  transportVehicles: TransportVehiclesInterface[];
}

File: backend/entity/transportcompanies.go
package entity

import (
	"time"
	"gorm.io/gorm"
)

type TransportCompanies struct {
	gorm.Model
    ID         uint      `gorm:"primaryKey"`
    TComName   string    `json:"company_name"`
    Contact    string    `json:"contact"`
    Email      string    `json:"email"`
    Address    string    `json:"address"`
    PostalCode string    `json:"postal_code"`
    StartDate  time.Time `json:"start_date"`
    EndDate    time.Time `json:"end_date"`
    Photo      string    `gorm:"type:longtext" json:"photocom"`
	TransportVehicles []TransportVehicle `gorm:"foreignKey:TransportCompaniesID"`
}